tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -788,25 +788,28 @@ Status StorageManager::load_fragment_metadata(
 }
 
 Status StorageManager::object_type(const URI& uri, ObjectType* type) const {
-  bool is_array;
-  RETURN_NOT_OK(this->is_array(uri, &is_array));
-  if (is_array) {
-    *type = ObjectType::ARRAY;
+  bool is_dir = false;
+  RETURN_NOT_OK(vfs_->is_dir(uri, &is_dir));
```

Comment:
For S3, this might not be necessary, since later on `RETURN_NOT_OK(vfs_->ls(uri, &uris));` would just yield an empty `uris` vector in case `uri` is not a directory. This would save one HTTP request per object in the directory. Special handling may be needed for {posix, win, hdfs} as `RETURN_NOT_OK(vfs_->ls(uri, &uris));` may return an error in case `uri` is not a directory (needs checking).